app/src/main/java/de/xikolo/storages/RecentCoursesStorage.kt

```diff
@@ -0,0 +1,51 @@
+package de.xikolo.storages
+
+import android.content.Context
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import de.xikolo.storages.base.BaseStorage
+
+typealias RecentCourse = Pair<String, String>
+
+const val MAX_SHORTCUTS = 4
+
+class RecentCoursesStorage : BaseStorage(PREF_RECENT_COURSES, Context.MODE_PRIVATE) {
+
+    companion object {
+        private const val PREF_RECENT_COURSES = "preference_recent_courses"
+
+        fun addCourseToRecentCourses(
+            courseId: String,
+            title: String,
+            courses: LinkedHashSet<RecentCourse>
+        ): LinkedHashSet<RecentCourse> {
+            return courses
+                // filter duplicates
+                .filter { it.first != courseId }
+                .toMutableSet()
+                // add visited course
+                .apply { this.add(Pair(courseId, title)) }
+                // take last MAX_SHORTCUT courses
+                .reversed()
+                .take(MAX_SHORTCUTS)
+                .reversed()
+                .toMutableSet() as LinkedHashSet<RecentCourse>
+        }
+    }
+
+    val recentCourses: LinkedHashSet<RecentCourse>
+        get() {
+            val json = getString(PREF_RECENT_COURSES)
+            val type = object : TypeToken<LinkedHashSet<RecentCourse>>() {}.type
```

Comment:
Does this work well with Kotlins typealiases? Because Gson is written for Java

Reply:
Probably typealiases are only syntax sugar and dissolved during compile time?